File: Lab3/task3.c
#include <stdio.h>
#define MAX 100

int table[MAX] = {0}; // Initializes to 0

int fiboM(int n)
{
    int arr[100];
    if (n == 1 || n == 2)
        return 1;
    else
    {
        if (table[n] == 0) // If table[n] = 0 that means the value is not in the table
            table[n] = fiboM(n - 1) + fiboM(n - 2);
        return table[n];
    }
}

int main()
{
    int n;
    printf("Enter the fibonacci term you want to find: ");
    scanf("%d", &n);
    int ans = fiboM(n);
    if (n == 1)
        printf("The %dst fibonacci term is %d", n, ans);
    else if (n == 2)
        printf("The %dnd fibonacci term is %d", n, ans);
    else if (n == 3)
        printf("The %drd fibonacci term is %d", n, ans);
    else
        printf("The %dth fibonacci term is %d", n, ans);
    return 0;
}

File: Lab3/task3.py
def fiboM(n):
    table = {}
    if n == 1 or n == 2:
        return 1
    else:
        if n not in table:
            result = fiboM(n-1) + fiboM(n-2)
            table[n] = result
        return table[n]
    

n = int(input("Enter the fibonacci term you want to find: "))
ans = fiboM(n)
if n == 1:
    print(f"The {n}st fibonacci term is {ans}")
elif n == 2:
    print(f"The {n}nd fibonacci term is {ans}")
elif n == 3:
    print(f"The {n}rd fibonacci term is {ans}")
else:
    print(f"The {n}st fibonacci term is {ans}")